File: go/clock/clock.go
package clock

import (
	"fmt"
)

type Clock struct {
	hour   int
	minute int
}

func (c Clock) FixTime() Clock {
	for c.minute < 0 {
		c.minute += 60
		c.hour--
	}

	for c.hour < 0 {
		c.hour += 24
	}

	for c.minute >= 60 {
		c.hour++
		c.minute -= 60
	}

	c.hour = c.hour % 24

	return c
}

func New(hour, minute int) Clock {
	var c Clock
	c.hour = hour
	c.minute = minute
	return c.FixTime()
}
func (c Clock) Add(minutes int) Clock {
	c.minute += minutes
	return c.FixTime()
}

func (c Clock) Subtract(minutes int) Clock {
	c.minute -= minutes
	return c.FixTime()
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

File: nim/clock/clock.nim
import strformat, math

type
  Clock* = object
    hour*: range[0..23]
    minute*: range[0..59]
  Minutes* = distinct int

proc initClock*(hour, minute: int): Clock =
  let totalMinutes = (hour * 60 + minute).euclMod(24 * 60)
  return Clock(hour: totalMinutes div 60, minute: totalMinutes mod 60)

proc `$`*(c: Clock): string =
  result = fmt"{c.hour:02}:{c.minute:02}"

proc `+`*(c: Clock, v: Minutes): Clock =
  return initClock(c.hour, c.minute + v.int)

proc `-`*(c: Clock, v: Minutes): Clock =
  return initClock(c.hour, c.minute - v.int)
